### PR TITLE
fix: cell padding and icon hit area

### PR DESCRIPTION
**Pull request by cbt1**
Changes the cell padding so that the left/ right cell padding is 8px and the top/bottom is 4px. As per design spec.

This also includes a change to the size of expand/collapse icon. To make it easier to hit, the icon now have a padding of 4px. This means that when clicking on the area, including the padding, the icons click handle is triggered.

The red area is the clickable area of the expand/collapse icon now. Previosly it was only the size of the actual icon. The orange area is the size of the `span` element with text. It now needs to have padding on it becase the container element can not have it any, as that would not work with the padding on the icon.
Skärmavbild 2023-10-31 kl  13 36 57

### CODE REVIEW

**Review by a-m-dev — APPROVED** [review]
Generally implementation looks good, 
Probably you already explained this, but AFAI understood , with hit area, we only supposed to have some extra padding between the (cell) text and the expand/collapse icon, but the rendering tests are seems a bit more than that, 
for example this one:
Screenshot 2023-11-01 at 16 59 44

or this one: 
Screenshot 2023-11-01 at 17 00 06


a bit of inspection over this, i realised that for a normal cell, now we have: 
cell padding, (if expand/collapse icon exists -> it's hit area) + grid gap + text left padding

https://github.com/qlik-oss/sn-pivot-table/assets/29652890/7cda9a37-b31a-4086-a2bf-1ad00d534c73

is it possible to skip (probably) grid gap and text padding in order to make the icon and text to be closer to each other

**Comment by cbt1** [comment]
> is it possible to skip (probably) grid gap and text padding in order to make the icon and text to be closer to each other

@a-m-dev 8px padding between the icon and the text is per design spec. So 8px on the cells left and right side and 8px between the icon and text.

File: src/pivot-table/hooks/use-column-width.ts [hunk 1]
```diff
@@ -214,12 +215,14 @@ export default function useColumnWidth(
...
+            estimateWidthForMeasureValue(qApprMaxGlyphCount) + TOTAL_CELL_PADDING,
+            measureTextForMeasureValue(qFallbackTitle) + TOTAL_CELL_PADDING,
           )
         : Math.max(
-            Math.max(...qMeasureInfo.map((m) => estimateWidthForMeasureValue(m.qApprMaxGlyphCount))),
-            estimateWidthForDimensionValue(qApprMaxGlyphCount) + leavesIconWidth,
+            Math.max(
+              ...qMeasureInfo.map((m) => estimateWidthForMeasureValue(m.qApprMaxGlyphCount) + TOTAL_CELL_PADDING),
+            ),
+            estimateWidthForDimensionValue(qApprMaxGlyphCount) + leavesIconWidth + TOTAL_CELL_PADDING,
```
[the remark this answers is not in the document]
Comment (cbt1) [reply src/pivot-table/hooks/use-column-width.ts:225]: Not really sure what you mean?

[the remark this answers is not in the document]
Comment (a-m-dev) [reply src/pivot-table/hooks/use-column-width.ts:225]: the `TOTAL_CELL_PADDING` is being added into all cases , so we can do something like 
```ts

const fitToContentWidth = (qApprMaxGlyphCount: number, qFallbackTitle: string) =>
      TOTAL_CELL_PADDING + topGridLeavesIsPseudo
        ? Math.max(
            estimateWidthForMeasureValue(qApprMaxGlyphCount),
            measureTextForMeasureValue(qFallbackTitle),
          )
        : Math.max(
            Math.max(
              ...qMeasureInfo.map((m) => estimateWidthForMeasureValue(m.qApprMaxGlyphCount)),
            ),
            estimateWidthForDimensionValue(qApprMaxGlyphCount) + leavesIconWidth,
```

[the remark this answers is not in the document]
Comment (cbt1) [reply src/pivot-table/hooks/use-column-width.ts:225]: Will fix 👍

**Pull request merged**